TeensyDmx.cpp

```diff
@@ -1686,6 +2141,58 @@ void TeensyDmx::handleByte(uint8_t c)
             maybeIncrementLengthMismatch();
             m_state = State::IDLE;
             break;
+        case State::RDM_DUB_PREAMBLE:
+            if (c == 0xAA) {
+              // We're not interested in DUB preamble bytes, so don't store them
+              // Start storing at index 8 so we can use DiscUniqueBranchResponse struct
+              m_dmxBufferIndex = 8;
+              m_state = State::RDM_DUB_RECV;
+            } else if (c == 0xFE) {
+              // Discarding bytes
+              // TODO(Peter): Check we only ever get 7 of these
+            } else {
+              // Unexpected preamble byte
+              m_state = State::IDLE;
+            }
+            break;
+        case State::RDM_DUB_RECV:
+            reinterpret_cast<uint8_t*>(&m_rdmBuffer)[m_dmxBufferIndex] = c;
+            ++m_dmxBufferIndex;
+            if (m_dmxBufferIndex >= (8+12)) {
+                m_state = State::RDM_DUB_CHECKSUM_3;
+            }
+            break;
+        case State::RDM_DUB_CHECKSUM_3:
+            m_rdmChecksum = (c << 8);
+            m_state = State::RDM_DUB_CHECKSUM_2;
+            break;
+        case State::RDM_DUB_CHECKSUM_2:
+            m_rdmChecksum = (m_rdmChecksum & (c << 8));
+            m_state = State::RDM_DUB_CHECKSUM_1;
+            break;
+        case State::RDM_DUB_CHECKSUM_1:
+            m_rdmChecksum = (m_rdmChecksum | c);
+            m_state = State::RDM_DUB_CHECKSUM_0;
+            break;
+        case State::RDM_DUB_CHECKSUM_0:
+            m_rdmChecksum = ((m_rdmChecksum & 0xff00) | ((m_rdmChecksum & 0x00ff) & c));
+            ++m_dmxBufferIndex;
+            if (m_rdmChecksum ==
+                    rdmCalculateChecksum(reinterpret_cast<uint8_t*>(&m_rdmBuffer) + 8,
```

Comment:
Magic numbers here, could we use sizeof?

Reply:
Done, had to do some pointer stuff as we want the sizeof a struct member.